### PR TITLE
add new heka json decoder that can mutate the message

### PR DESCRIPTION
**Pull request by relud**
@trink

### CODE REVIEW

File: heka/io_modules/decoders/heka/json_mutate.lua [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+local pairs          = pairs
+local type           = type
+
+local cfg = read_config(module_cfg) or {}
+assert(type(cfg) == "table", module_cfg .. " must be a table")
+cfg.scrub_fields = cfg.scrub_fields or {}
+cfg.user_agent_transforms = cfg.user_agent_transforms or {}
+cfg.flatten_delimiter = cfg.flatten_delimiter or "."
+assert(type(cfg.flatten_delimiter) == "string", module_cfg .. ".flatten_delimiter must be a string")
+fields_prefix = "Fields" .. cfg.flatten_delimiter
```
Comment (trink) [inline heka/io_modules/decoders/heka/json_mutate.lua]: Needs to be local

Comment (relud) [inline heka/io_modules/decoders/heka/json_mutate.lua]: fixed

**Comment by trink** [comment]
This needs to be against the dev branch like the original PR

**Comment by trink** [comment]
I pulled the branch down and merged it into dev so it will make this weeks packaging

**Comment by relud** [comment]
ah, thank you and my bad

**Pull request closed without merging**